worker-jobtracking-container/src/test/java/com/hpe/caf/worker/jobtracking/JobTrackingWorkerIT.java

```diff
@@ -98,15 +101,60 @@ public void testTrackingReportTasks() throws Exception
         jobDatabase.createJobTask(defaultPartitionId, jobTaskId, "testProxiedActiveMessage");
         for (int i = 0; i <= 4; i++) {
             final JobStatus status = i < 4 ? JobStatus.Waiting : JobStatus.Completed;
+            final int percentageCompleted = i < 4 ?  0 : 100;
             final TrackingReportStatus trackingReportStatus = i < 4 ? TrackingReportStatus.Progress : TrackingReportStatus.Complete;
-            final TaskMessage taskMessage = getExampleTrackingReportMessage(defaultPartitionId, jobTaskId, trackingReportStatus);
-            final JobReportingExpectation expectation = getExpectation(jobTaskId, status);
+            final TaskMessage taskMessage = getExampleTrackingReportMessage(defaultPartitionId, jobTaskId, trackingReportStatus,
+                    percentageCompleted);
+            final JobReportingExpectation expectation = getExpectation(jobTaskId, status, percentageCompleted);
             publishAndVerify(taskMessage, jobTaskId, expectation);
         }
     }
 
-    private JobReportingExpectation getExpectation(final String jobTaskId, final JobStatus status){
-        final int percentageCompleted = status.equals(JobStatus.Completed) ? 100 : 0;
+    @Test
+    public void testTrackingReportProgressTasks() throws Exception
+    {
+        final String jobTaskId = jobDatabase.createJobId();
+        jobDatabase.createJobTask(defaultPartitionId, jobTaskId, "testProxiedActiveMessage");
+        for (int i = 0; i <= 4; i++) {
+            final JobStatus status = (i == 0) ? JobStatus.Waiting : JobStatus.Active;
+            final int percentageCompleted = ((25*i) == 100) ? 99 : (25*i);
+            final TaskMessage taskMessage = getExampleTrackingReportMessage(defaultPartitionId, jobTaskId,
+                    TrackingReportStatus.Progress, percentageCompleted);
+            final JobReportingExpectation expectation = getExpectation(jobTaskId, status, percentageCompleted);
+            publishAndVerify(taskMessage, jobTaskId, expectation);
+        }
+    }
+
+    @Test
+    @Description("Throws an exception if taskId is empty or null")
+    public void testProgressReportingMessagesEmptyTaskId() {
+        final Exception exception1 = assertThrows(SQLException.class,
+                ()->jobDatabase.reportTaskProgress(defaultPartitionId, "", 18));
+        final Exception exception2 = assertThrows(SQLException.class,
+                ()->jobDatabase.reportTaskProgress(defaultPartitionId, null, 18));
+
+        final String expectedMessage = "Task identifier has not been specified";
+
+        Assert.assertTrue(exception1.getMessage().contains(expectedMessage));
+        Assert.assertTrue(exception2.getMessage().contains(expectedMessage));
+    }
+
+    @Test
+    @Description("Throws an exception if percentage_complete is invalid")
```

Comment:
Is this preferred over using JavaDoc in some situations?  Does it display the description when the test fails or something like that?  I don't think I've seen it before.  Where did you come across it?

Reply:
Sorry, just habit I had, using @Description or @DisplayName (org.junit.jupiter.api) to add a quickly readable description of what's being done. It makes it easier in the test view on the IDE. But as the convention on that project is javadoc, I ll replace with it. Also, I tested and surefire doesn't pick any DisplayName, javadoc or Description. Opening the javadoc, we can see DisplayName and @Description but a little less clean than classic javadoc, especially if you don't know they exist.